sklearn/feature_extraction/text.py

```diff
@@ -909,6 +911,10 @@ def _make_int_array():
     """Construct an array.array of a type suitable for scipy.sparse indices."""
     return array.array(str("i"))
```

Comment:
BTW, the str() was there from before, so presumably someone else had the need for the conversion at some earlier point too.

The Python where this failed for me is a UCS-2 build of Python 2.7.10 on a 64-bit OS X system:

Python 2.7.10 (default, Oct 23 2015, 18:05:06) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)] on darwin